### DISCUSSION TITLE
Stacey Boundary conditions - SPECFEM2D GPU

### DISCUSSION THREAD

**Discussion by etotten93** [SPECFEM/specfem3d#1736] [General]
Hi There,

I have recently started using SPECFEM2D GPU (fortran) to great effect (thanks to the developers and maintainers) - I notice that PML ABCs are not implemented when running on GPU, so I have had to fall back to Stacey boundary conditions. At present, they are not proving very effective (see attached snapshots on a test run).

So my questions are as follows:
1) Is there a way to increase/up the strength of the Stacey ABCs in the Par_file/similar
2) If not, is the next best practice to pad the left, bottom and right (top is free surface) boundary with a thick layer of uniform velocity such that any artificial/numerical reflections never hit any seismometer receiver (this is my priority)?

Many thanks in advance for your time and help!

Eoghan

![forward_image000001600](https://github.com/user-attachments/assets/97f2378a-1868-4fd6-981b-f7df44192ecb)
![forward_image000003400](https://github.com/user-attachments/assets/fb86eb8d-6080-4830-bf60-efb093c97b3b)
![forward_image000010400](https://github.com/user-attachments/assets/a3d57d67-708d-4e31-8a10-a01aa4bfa925)

**Comment by danielpeter** [comment]
hi Eoghan, 

yes, for GPU simulations only Stacey is provided for now. thus, the work-around is as you suggested, i.e., enlarge the domain to make sure that the boundary reflections hit your stations after the time of interest.
